share/disambiguation.py

```diff
@@ -145,19 +144,25 @@ class AbstractCreativeWorkDisambiguator(Disambiguator):
     FOR_MODEL = AbstractCreativeWork
 
     def disambiguate(self):
-        if self.attrs.get('links'):
-            for link in self.attrs.get('links'):
-                models = ThroughLinks.objects.select_related('creative_work', 'link').filter(link=link)[:2].all()
-                if len(models) == 1 and 'issn' not in models[0].link.type.lower():
-                    return models[0].creative_work
+        if self.attrs.get('identifiers'):
+            for id in self.attrs.get('identifiers'):
+                try:
+                    identifier = Identifier.objects.get(id=id)
+                    return identifier.abstractcreativework_set.first()
```

Comment:
Ideally, yes, but (at the moment) it's possible to end up with multiple. Like if two works exist, with identifiers `I1` and `I2`, and then we harvest a work with identifiers `I1` and `I2`. Right now the new work might disambiguate on `I1`, but then `I2` will point to two works. I'm not sure if there's a better way... maybe it'd be better to merge the two existing works and then update with the new one, but we need to fix merging first, at least.